### ISSUE TITLE
It crashses giving the following error

### ISSUE THREAD

**Issue by dark-warlord14** [hakluke/hakcheckurl#7]
```
goroutine 408 [running]:                                                
main.main.func1(0xc00001a340, 0xc00001e180, 0xc000068ba0, 0xc0000991d0, 0x46)                                                                         
        /root/go/src/github.com/hakluke/hakcheckurl/main.go:51 +0xe5                                                                                  
created by main.main                                                                                                                                  
        /root/go/src/github.com/hakluke/hakcheckurl/main.go:46 +0x2df
                                                                                                                                                      
                                                                                                                                                      
panic: runtime error: invalid memory address or nil pointer dereference                                                                               
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x62f8c5]

```

**Comment by hakluke** [comment]
Hey @dark-warlord14 can you give me the exact command you ran?

Also ensure that you are using the most recent version.

**Comment by dark-warlord14** [comment]
I used the most basic one
`waybackurls -no-subs yahoo.com | hakcheckurl > result.txt`
May be it is due handling of goroutines. I dont know much about go but the crash happens if you feed the tool a vast amount of data.

**Comment by Ice3man543** [comment]
@hakluke this seems to be caused by assigning headers before checking if the request has been created successfully. A moving the error check up would solve the issue.

https://github.com/hakluke/hakcheckurl/blob/831a993f088daebdfaf3a0a532dfd4f332364e4a/main.go#L50-L55

**Comment by pczajkowski** [comment]
Should've been fixed by 014c9fc.

**Comment by hakluke** [comment]
Done!
